### DISCUSSION TITLE
Add plugins to docker without creating new image?

### DISCUSSION THREAD

**Discussion by macmandr197** [verdaccio/verdaccio#2042] [Q&A] [answered]
Hey guys, saw a couple examples where you had a plugin (I think it was the memory plugin?) inside the "plugins" folder for your docker container so that you could load the plugin into the active container without having to create a whole new image. I'm currently using docker within unraid and have my plugins folder mapped out. What I'm trying to do now is add the Github oauth plugin [https://github.com/n4bb12/verdaccio-github-oauth-ui](url) to my docker container so it can use github authentication rather than the default htpassword file. Any ideas?

**Comment by juanpicado** [comment, marked as answer]
~Not possible at this point.~ sorry wrong answer.

Here you can use `docker-compose` example

https://verdaccio.org/docs/docker#using-docker-compose

or mapping directly in the terminal

```
V_PATH=/path/for/verdaccio; docker run -it --rm --name verdaccio \
  -p 4873:4873 \
  -v $V_PATH/conf:/verdaccio/conf \
  -v $V_PATH/storage:/verdaccio/storage \
  -v $V_PATH/plugins:/verdaccio/plugins \
  verdaccio/verdaccio
```  

Yo need map the config file, because you need to update which plugins will use and the plugin folder, you could just clone [verdaccio-github-oauth-ui](https://github.com/verdaccio/verdaccio/discussions/url) or copy the content of the plugin there (not very clean solution) but I recommend better create your own docker image, is much reliable. 


https://verdaccio.org/docs/docker#adding-plugins-without-creating-a-new-image
